### PR TITLE
Add a unit test for `atoms.core.pertub`

### LINKED ISSUE #2048: Add a test for `atoms.core.perturb`
### What would you like to report?

https://github.com/Quantum-Accelerators/quacc/blob/0b25767275d8a9de5a863807a41cef2a5ae79489/src/quacc/atoms/core.py#L260-L278

This function should have a dedicated unit test.

### PR DESCRIPTION
**Pull request by mhbarrie**
## Summary of Changes

Added unit test for function perturb in quacc/src/quacc/atoms/core.py

Closes #2048.

### Checklist

- [X ] I have read the ["Guidelines" section](https://quantum-accelerators.github.io/quacc/dev/contributing.html#guidelines) of the contributing guide. Don't lie! 😉
- [ X] My PR is on a custom branch and is _not_ named `main`.
- [X ] I have added relevant, comprehensive [unit tests](https://quantum-accelerators.github.io/quacc/dev/contributing.html#unit-tests).

### CODE REVIEW

**Comment by buildbot-princeton** [comment]
Can one of the admins verify this patch?

File: tests/core/atoms/test_atoms_core.py [hunk 1]
```diff
@@ -208,3 +213,18 @@ def test_check_charge_and_spin(os_atoms):
...
+    scale = 0.5
+    perturbed_atoms = perturb(atoms, matrix, scale)
+    assert np.allclose(
+        perturbed_atoms.get_positions(),
+        [[0.05, 0.05, 0.05], [0.1, 0.1, 0.84]],
+        atol=1e-3,
+    ), "Perturbation did not work as expected"
+    assert np.allclose(
+        atoms.get_positions(), [[0, 0, 0], [0, 0, 0.74]], atol=1e-3
+    ), "Original atoms object was modified"
```
Comment (Andrew-S-Rosen) [inline tests/core/atoms/test_atoms_core.py:230]: I'd recommend using the dedicated numpy testing utility for this, namely [`assert_allclose`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html).

It'd be used as

```python
from numpy.testing import assert_allclose

assert_allclose(...) # no need to put an `assert` in front
```

File: tests/core/atoms/test_atoms_core.py [hunk 2]
```diff
@@ -208,3 +213,18 @@ def test_check_charge_and_spin(os_atoms):
...
+    matrix = [[0.1, 0.1, 0.1], [0.2, 0.2, 0.2]]
+    scale = 0.5
+    perturbed_atoms = perturb(atoms, matrix, scale)
+    assert np.allclose(
+        perturbed_atoms.get_positions(),
+        [[0.05, 0.05, 0.05], [0.1, 0.1, 0.84]],
+        atol=1e-3,
+    ), "Perturbation did not work as expected"
+    assert np.allclose(
+        atoms.get_positions(), [[0, 0, 0], [0, 0, 0.74]], atol=1e-3
```
Comment (Andrew-S-Rosen) [inline tests/core/atoms/test_atoms_core.py:229]: I'd suggest leaving the default `atol` if possible.

File: tests/core/atoms/test_atoms_core.py [hunk 3]
```diff
@@ -208,3 +213,18 @@ def test_check_charge_and_spin(os_atoms):
     charge, spin_multiplicity = check_charge_and_spin(atoms, charge=0)
     assert charge == 0
     assert spin_multiplicity == 3
+
+
+def test_perturb():
```
Comment (Andrew-S-Rosen) [inline tests/core/atoms/test_atoms_core.py:218]: Could you add one additional check to make sure that the original `atoms` object is unmodified after calling `perturb`? Thanks!

**Review by Andrew-S-Rosen — COMMENTED** [review]
Thank you very much @mhbarrie for this timely PR! I greatly appreciate it. I have left a few minor comments below that would be great if you could address.

**Comment by mhbarrie** [comment]
Will try and make those changes soon currently working on issue 1905.

**Comment by Andrew-S-Rosen** [comment]
Thank you, @mhbarrie!! I'm setting this to auto-merge.

**Pull request merged**